### PR TITLE
add open-telemetry sample

### PR DESCRIPTION
**Pull request by zirain**
**Please provide a description of this PR:**

add open-telemetry sample
related PR: https://github.com/istio/istio/pull/36817

### CODE REVIEW

File: samples/open-telemetry/README.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
+# Open Telemetry ALS
+
+This sample demonstrates Istio's Open Telemetry ALS support.
+
+## Start otel-collector service
+
+First, we need to create an `otel-collector` backend with simple configuration.
```
Comment (ericvn) [inline samples/open-telemetry/README.md:7]:
```suggestion
First, create an `otel-collector` backend with simple configuration.
```

**Review by ericvn — COMMENTED** [review]
Some initial comments

**Comment by ericvn** [comment]
On other comment is I typically put a space between the -n and the namespace in kubectl commands. I see the commands here do not. Trying the command on my system it appears that the space is not needed.

**Comment by zirain** [comment]
> needed

it's not needed, add one will be better.

**Comment by zirain** [comment]
ping @ericvn

**Comment by ericvn** [comment]
One final comment. Do we want to create a release note announcing this sample was created?

**Comment by zirain** [comment]
> One final comment. Do we want to create a release note announcing this sample was created?

there's a release note https://github.com/istio/istio/pull/36817, do we need another for the doc?

**Review by ericvn — APPROVED** [review]

**Comment by hzxuzhonghu** [comment]
@zirain Better write a user guide to istio.io

**Comment by stevenctl** [comment]
@zirain I'm assuming we want this in 1.13?

**Pull request merged**